### PR TITLE
chore: tagoverflow typescript conversion

### LINKED ISSUE #5673: Add Typescript types to TagOverflow

### PR DESCRIPTION
**Pull request by davidmenendez**
Closes #5673

### CODE REVIEW

File: packages/ibm-products/src/components/TagOverflow/TagOverflowPopover.tsx [hunk 1]
```diff
@@ -5,83 +5,73 @@
...
-          ? index < allTagsModalSearchThreshold
-          : index <= allTagsModalSearchThreshold
+      const thresh = allTagsModalSearchThreshold ?? 0;
+      return overflowTags?.filter((_, index) =>
+        overflowTags?.length > thresh ? index < thresh : index <= thresh
       );
     };
 
+    const visibleItems = getOverflowPopoverItems();
+    const hasItems = visibleItems && visibleItems?.length > 0;
```
Comment (amal-k-joy) [inline packages/ibm-products/src/components/TagOverflow/TagOverflowPopover.tsx:64]: same comment as above
`const hasItems = visibleItems?.length > 0;`

**Review by amal-k-joy — CHANGES_REQUESTED** [review]
Good work 
Please see some minor comments

**Review by amal-k-joy — APPROVED** [review]

File: packages/ibm-products/src/components/TagOverflow/TagOverflow.tsx [hunk 2]
```diff
@@ -5,78 +5,123 @@
...
 import { TagOverflowPopover } from './TagOverflowPopover';
 import { TagOverflowModal } from './TagOverflowModal';
 
+export interface TagOverflowItem {
+  className?: string;
+  filter?: string;
+  id: string;
+  label: string;
+  onClose: () => void;
+  tagType?:
```
Comment (matthewgallo) [inline packages/ibm-products/src/components/TagOverflow/TagOverflow.tsx:36]: If possible we should try to import this from `@carbon/react`.

**Review by matthewgallo — APPROVED** [review]
Found some types that we might be able to just pull in from `@carbon/react`, but we can open a separate issue.

**Pull request merged**